### PR TITLE
Fix some issues with __round__

### PR DESCRIPTION
**Pull request by msullivan**
In python 3, add the overload for there being no digits argument,
(which was wrong, and causing mypy to miss that things were
SupportsFloat (though that itself might be a bug?)).

In python 2, __round__ doesn't exist and SupportsRound doesn't exist
in the typing module. Use SupportsFloat for python 2 round().

### CODE REVIEW

File: stdlib/3/typing.pyi [hunk 1]
```diff
@@ -95,6 +95,8 @@ class SupportsAbs(Protocol[_T_co]):
 
 @runtime
 class SupportsRound(Protocol[_T_co]):
+    @abstractmethod
+    def __round__(self) -> _T_co: ...
```
Comment (ilevkivskyi) [inline stdlib/3/typing.pyi]: Wait, why is this needed? `ndigits` is already an optional argument.

Comment (msullivan) [reply stdlib/3/typing.pyi]: The first version of this PR was busted and only passed testing because I introduced errors that allowed it to pass <_<.

In the just-pushed updated version, the overload is fixed to properly specialize the type behavior, which is that it should return `int` when called without ndigits. (This matches the signature of `round` and the signatures of `__round__` on concrete types.)

File: stdlib/2and3/decimal.pyi [hunk 2]
```diff
@@ -124,8 +124,6 @@ class Decimal(object):
         @overload
         def __round__(self) -> int: ...
         @overload
-        def __round__(self, ndigits: None) -> int: ...
-        @overload
```
Comment (srittau) [inline stdlib/2and3/decimal.pyi]: In my defense, the pure Python implementation of `Decimal.__round__()` does support `None`. But it seems the C version doesn't. Good catch.
